### PR TITLE
Fix undefined array key

### PR DESCRIPTION
**Pull request by Diguzz**

### CODE REVIEW

File: src/market.php [hunk 1]
```diff
@@ -184,7 +184,7 @@
 
 			$filtrobusca = "" . $orderby . " " . $sort . "";
 
-			$query = $db->execute("select market.market_id, market.price, market.seller, blueprint_items.name, blueprint_items.effectiveness, blueprint_items.needpromo, blueprint_items.needring, blueprint_items.voc, blueprint_items.img, items.item_bonus, items.for, items.vit, items.agi, items.res from `market`, `blueprint_items`, `items` where market.ite_id=blueprint_items.id and market.market_id=items.id and blueprint_items.type=? and market.serv=? order by " . $filtrobusca . "", [$_GET['type'], $player->serv, $orderby]);
+			$query = $db->execute("select market.market_id, market.price, market.seller, blueprint_items.name, blueprint_items.effectiveness, blueprint_items.needpromo, blueprint_items.needring, blueprint_items.voc, blueprint_items.img, items.item_bonus, items.for, items.vit, items.agi, items.res from `market`, `blueprint_items`, `items` where market.ite_id=blueprint_items.id and market.market_id=items.id and blueprint_items.type=? and market.serv=? order by " . $filtrobusca . "", [isset($_GET['type']), $player->serv, $orderby]);
```
Comment (juniorb2ss) [inline src/market.php:187]: `isset($_GET['type']` will return an boolean, not the value, invalidading the query.

File: src/work.php [hunk 2]
```diff
@@ -73,7 +73,7 @@
 	$ganha = 20 * (1 + $bonnus);
 }
 
-if ($_GET['act'] == "cancel") {
+if (isset($_GET['act']) == "cancel") {
```
Comment (juniorb2ss) [inline src/work.php:76]: isset will return an boolean, not the strict value.

File: src/members.php [hunk 3]
```diff
@@ -58,15 +58,15 @@
...
+} elseif (isset($_GET['orderby']) == "level") {
 	$selecum = "selected";
-} elseif ($_GET['orderby'] == "gold") {
+} elseif (isset($_GET['orderby']) == "gold") {
 	$selecdois = "selected";
-} elseif ($_GET['orderby'] == "kills") {
+} elseif (isset($_GET['orderby']) == "kills") {
 	$selectres = "selected";
-} elseif ($_GET['orderby'] == "monsterkilled") {
+} elseif (isset($_GET['orderby']) == "monsterkilled") {
```
Comment (juniorb2ss) [inline src/members.php:69]: isset will return an boolean, not the strict value.

File: src/battle.php [hunk 4]
```diff
@@ -1034,7 +1034,7 @@
 		echo "<tr><td></td><td><br /><input type=\"submit\" value=\"Procurar\" /></td></tr>\n";
 		echo "</table>\n";
 		echo "</form>\n</fieldset>\n";
-		if (!$_GET['reino'] || $_GET['reino'] == $player->reino) {
+		if (!isset($_GET['reino']) || isset($_GET['reino']) == $player->reino) {
```
Comment (juniorb2ss) [inline src/battle.php:1037]: isset will return an boolean, not the strict value.

File: src/chat.php [hunk 5]
```diff
@@ -165,7 +165,7 @@ function startChatSession(): void
 ?>
 	{
 	"username": "<?php echo str_replace(" ", "_", $player->username); ?>",
-	"level": "<?php echo $_SESSION['level']; ?>",
+	"level": "<?php echo isset($_SESSION['level']); ?>",
```
Comment (juniorb2ss) [inline src/chat.php:168]: isset will return an boolean, not the strict value.

**Review by juniorb2ss — CHANGES_REQUESTED** [review]
In a lot of place `isset()` is being used and later comparing strict value, which is wrong, as isset will return an boolean, not the strict value.

**Comment by Diguzz** [comment]
I misunderstood the isset() function, thinking it returned the variable's value rather than just checking if it’s set. I'll soon post the corrected code addressing this misunderstanding. thanks for the feedback.

**Pull request closed without merging**